modules/core/src/main/kotlin/datamaintain/core/step/Filter.kt

```diff
@@ -12,7 +12,7 @@ class Filter(private val context: Context) {
     fun filter(scripts: List<ScriptWithContent>): List<ScriptWithContent> {
         try {
             if (!context.config.porcelain) { logger.info { "Filter scripts..." } }
-            var filteredScripts = scripts
+            var filteredScripts = scripts.filter { context.config.filenameRegex.matches(it.name) }
```

Comment:
Maybe it could be useful to add a method that filters based on a criteria and logs a given message when the script was filtered. Its prototype could be:
```kotlin
fun filterBasedOnCriteria(filterCriteria: Function<ScriptWithContent, Boolean>, messageToLogUponFilter: String): List<ScriptWithContent>
```